opensearch-service-domain-cdk/lib/stack-composer.ts

```diff
@@ -0,0 +1,48 @@
+import { Construct } from 'constructs';
+import { Stack, StackProps } from 'aws-cdk-lib';
+import {OpensearchServiceDomainCdkStack} from "./opensearch-service-domain-cdk-stack";
+import {EngineVersion} from "aws-cdk-lib/aws-opensearchservice";
+
+export class StackComposer {
+
+    constructor(scope: Construct, props: StackProps) {
+
+        let version: EngineVersion
+
+        const domainName: string = scope.node.tryGetContext('domainName')
+        const dataNodeType: string = scope.node.tryGetContext('dataNodeType')
+        const dataNodeCount: number = scope.node.tryGetContext('dataNodeCount')
+        const dedicatedMasterNodeType: string = scope.node.tryGetContext('dedicatedMasterNodeType')
+        const dedicatedMasterNodeCount: number = scope.node.tryGetContext('dedicatedMasterNodeCount')
+        const warmNodeType: string = scope.node.tryGetContext('warmNodeType')
+        const warmNodeCount: number = scope.node.tryGetContext('warmNodeCount')
+
+        const engineVersion: string = scope.node.tryGetContext('engineVersion')
+        if (engineVersion.startsWith("OS_")) {
+            // Will accept a period delimited version string (i.e. 1.3) and return a proper EngineVersion
+            version = EngineVersion.openSearch(engineVersion.substring(3))
+        }
+        else if (engineVersion.startsWith("ES_")) {
+            version = EngineVersion.elasticsearch(engineVersion.substring(3))
+        }
+        else {
+            throw new Error("Engine version is not present or does not match the expected format, i.e. OS_1.3 or ES_7.9")
+        }
+
+        const opensearchStack = new OpensearchServiceDomainCdkStack(scope, 'OpenSearchServiceDomainCDKStack', {
```

Comment:
It's generally best to embed things like stage (alpha/beta/gamma/prod) and region into top-level stack names (like `'OpenSearchServiceDomainCDKStack'` here) as it helps w/ naming collisions when people deploy multiple into the same account.  Things like IAM Roles are global to the account and the CDK usually creates their names using the stack name and/or higher-level resource name as a suffix.  This means if you deploy the same stack to two regions, and the CDK Stack has a region-agnostic name, you can end up with collisions on global resources even though the stacks themselves are deployed to separate regions.

Reply:
Definitely a valid point, I will try to shape these resource names to incorporate this